Improving seed data

yock opened the pull request:
<!--Read comments, before commiting pull request read checklist again

# Checklist:

- I have performed a self-review of my own code,
- I have commented my code, particularly in hard-to-understand areas,
- I have made corresponding changes to the documentation,
- I have added tests that prove my fix is effective or that my feature works,
- New and existing unit tests pass locally with my changes ("bundle exec rake"),
- Title include "WIP" if work is in progress.

-->

Resolves #157

### Description

Seeds data for Several wishlists, including site managers, real data for 20 Amazon products, and some owned and anonymous pledges.

### Type of change

* New feature (non-breaking change which adds functionality)

### How Has This Been Tested?

Existing test suite passes. My own testing consisted of running the seeds, running the app, browsing and clicking through to products, and clicking through to create new pledges.

### Screenshots
<img width="1240" alt="playtime" src="https://user-images.githubusercontent.com/306343/31869610-1c5ed6be-b777-11e7-8429-23e2626e2899.png">

Commit:
Improving seed data

leesharma commented on db/seeds.rb:
```diff
@@ -1,34 +1,43 @@
+    WishlistItem.create!(
+      quantity: 1,
+      wishlist: wishlist,
+      item: items.sample
```
There's a small bug here: sampling allows for duplicates. We probably want something like `list.shuffle.take(n)` instead.

![duplicate sampled items](https://user-images.githubusercontent.com/814638/31872603-31bde950-b78b-11e7-97a8-13b4a02ed55f.png)

While we're here, could you invert this assignment (i.e. randomly assign an item to several wishlists)? That way, the wishlist size will vary a bit and all of the items will be used.

leesharma reviewed (changes requested):
Wow, awesome work. Thanks for the PR! Breaking this into multiple files is a nice touch. 🙂 

I've got a couple small requests, but this looks great overall! Let me know if you have any questions/comments/disagreements.

Commit:
Seed admin user

Commit:
Put deletes in dependent order

In Rails, `delete_all` is faster than `destroy_all`. To keep things
fast, this reorders the destroy operations and turns them into deletes.

Commit:
Assign each item to random wishlists

Inverts the logic from wishlist -> items to items -> wishlists. This
ensures that all items are assigned and that no wishlist can have the
same item twice.

yock commented:
@leesharma After inverting the assignment loop I still see duplicates on the home page of the site when I run it locally. Would this be expected if an items appears in multiple wishlists? If so, might this be desired behavior or a bug?

leesharma commented:
@yock I see what you mean. The seeds have the desired behavior, but we'll definitely need to address that in the UI. I'll add a separate issue for that.

This looks good to me! :+1: Any other changes you want to make, or is this ready to merge?

yock commented:
@leesharma It's ready from my perspective. 👍

leesharma commented:
Perfect–thank you! I especially appreciated the astronauts. 🚀
